### ISSUE TITLE
[Feature Request] Add warning for docker interactive mode

### ISSUE THREAD

**Issue by Nkmol** [koalaman/shellcheck#2083]
#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

docker exec container_1 bash << 'HEREDOC'
        ls -l
HEREDOC

```

#### Here's what shellcheck currently says:
"No issues detected!"

#### Here's what I wanted or expected to see:
I want shellcheck to warn about not using docker interactive mode (`-i`). This simply is not want you would expect. Enabling interactive mode is needed in this case to the keep STDIN open.

This would be expected for all of the redirect operators, like:

- `echo "ls -l" | docker exec container_1 bash`
- `docker exec container_1 other_command < myfile.txt`
